Dockerfile

```
@@ -1,6 +1,10 @@
 # Use a Python image with uv pre-installed
 FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim
 
+# Setup a non-root user
+RUN addgroup --gid 10001 --system nonroot \
+ && adduser  --uid 10000 --system --ingroup nonroot --home /home/nonroot nonroot
```

I've seen conventionally system accounts in docker you'd use gid/uid 999 via groupadd/useradd

```suggestion
RUN groupadd --system --gid 999 nonroot \
 && useradd --system --gid 999 --uid 999 --create-home nonroot
```

---

Some examples

* https://github.com/redis/docker-library-redis/blob/a653c31bc10b9d591a9598886138a0c8a0bffd48/7.4/debian/Dockerfile#L10-L12
* https://github.com/docker-library/postgres/blob/a2433755c76d294477c85945d68944f8cdb7cf4b/18/trixie/Dockerfile#L10-L13

---

No particular technical downside besides being very unconventional